### ISSUE TITLE
PHP Fatal error: Call to undefined function ini_set

### ISSUE THREAD

**Issue by maltfield** [dompdf/dompdf#3523]
This bug report fixes a PHP Fatal error caused by `Dompdf.php` when `ini_set` is disabled

# Problem

There is a bug in `vendor/dompdf/dompdf/src/Dompdf.php at the following lines:

```
        @ini_set('pcre.jit', '0');
```

and

```
            @ini_set('pcre.jit', $this->pcreJit);
```

These lines causes a PHP Fatal error on hardened systems with the `ini_set` function disabled.

```
PHP Fatal error:  Uncaught Error: Call to undefined function Dompdf\\ini_set() in /path/to/wp-content/plugins/woocommerce-pdf-invoices-packing-slips/vendor/dompdf/dompdf/src/Dompdf.php:293...
```

# Why this matters

For security reasons, orgs frequently configure php.ini to be hardened by adding many dangerous functions to the `disable_functions` variable in the php.ini file. For example, it's common to disable the 'exec' function

```
disable_functions = exec
```

Of course, if a php script could modify the php configuration, then it would defeat any hardening done by setting `disable_functions`. As such, it's common to add `ini_set` to the `disable_functions`

```
disable_functions = exec, ini_set
```

# Solution

To fix the PHP Fatal error, we should always check to see if the `ini_set` function exists before attempting to call it

```
if( function_exists( 'ini_set') ){
        @ini_set('pcre.jit', '0');
}
```

# Context

This error was discovered on a server running the latest version of all software on the latest version of Debian, at the time of writing.

```
root@host:~# cat /etc/issue
Debian GNU/Linux 12 \n \l

root@host:~# 

root@host:~# dpkg -l | grep -iE 'php-common'
ii  php-common                      2:93                                    all          Common files for PHP packages
root@host:~# 
```

**Comment by maltfield** [comment]
See also downstream bug report:

 * https://github.com/dompdf/dompdf/issues/3523

**Comment by maltfield** [comment]
Submitted PR to fix this:

 * https://github.com/dompdf/dompdf/pull/3524

**Comment by maltfield** [comment]
This appears to be a regression bug?

 * https://github.com/dompdf/dompdf/pull/2049

**Comment by bsweeney** [comment]
> This appears to be a regression bug?

Not a regression, the original fix prefixed those calls with an `@` to suppress errors and that change is still present.

Are you sure the error is from Dompdf? Your original message references PEAR.php:52.

**Comment by maltfield** [comment]
Yes. I opened a bug report with both PEAR and Dompdf. Sorry , copy and paste error. I fixed it above.

**Comment by maltfield** [comment]
> Not a regression, the original fix prefixed those calls with an @ to suppress errors and that change is still present.

Actually, I don't think that would even fix this issue. I also reported a similar bug to Wordpress.

It seems that what changed is that, since PHP 8, this no longer just throws an error: it triggers a Fatal Error. So it's much more severe and prepending an `@` won't work.

In this case, `Dompdf` is being used by a wordpress plugin, and this bug breaks the whole wordpress site.
